src/buffer/out/textBuffer.cpp

```diff
@@ -1033,238 +1036,172 @@ const TextBuffer::TextAndColor TextBuffer::GetTextForClipboard(const bool lineSe
-        szSpanFontSize.resize(cbSpanFontSize); //chop off null at end
-
-        std::string const szSpanStartPattern = R"X(<SPAN STYLE="color:#%02x%02x%02x;background-color:#%02x%02x%02x">)X";
-
-        size_t const cbSpanStart = 53; // when format is expanded, there will be 53 bytes per color pattern.
-        std::string szSpanStart;
-        szSpanStart.resize(cbSpanStart + 1); // +1 for null terminator
+        std::ostringstream htmlBuilder;
 
-        std::string const szSpanStartFontPattern = R"X(<SPAN STYLE="font-family: '%s', monospace">)X";
-        size_t const cbSpanStartFontPattern = 41;
-
-        std::string const szSpanStartFontConstant = R"X(<SPAN STYLE="font-family: monospace">)X";
-        size_t const cbSpanStartFontConstant = 37;
+        // First we have to add some standard
+        // HTML boiler plate required for CF_HTML
+        // as part of the HTML Clipboard format
+        constexpr std::string_view HtmlHeader =
+            "<!DOCTYPE><HTML><HEAD><TITLE>Windows Terminal</TITLE></HEAD><BODY>";
+        htmlBuilder << HtmlHeader;
 
-        std::string szSpanStartFont;
-        size_t cbSpanStartFont;
-        bool fDeleteSpanStartFont = false;
+        htmlBuilder << "<!--StartFragment -->";
 
-        std::wstring const wszFontFaceName = fontFaceName;
-        size_t const cchFontFaceName = wszFontFaceName.size();
-        if (cchFontFaceName > 0)
+        // apply global style in div element
         {
-            // measure and create buffer to convert face name to UTF8
-            int const cbNeeded = WideCharToMultiByte(CP_UTF8, 0, wszFontFaceName.data(), static_cast<int>(cchFontFaceName), nullptr, 0, nullptr, nullptr);
-            std::string szBuffer;
-            szBuffer.resize(cbNeeded);
+            htmlBuilder << "<DIV STYLE=\"";
+            htmlBuilder << "display:inline-block";
+            htmlBuilder << "width:400px;";
+            htmlBuilder << "white-space:pre;";
+
+            // fixme: this is only walkaround for filling background after last char of row.
+            // It is based on first char of first row, not the actual char at correct position.
+            htmlBuilder << "background-color:";
+            const COLORREF globalBgColor = rows.BkAttr.at(0).at(0);
+            htmlBuilder << Utils::ColorToHexString(globalBgColor);
+            htmlBuilder << ";";
+
+            htmlBuilder << "font-family:'";
+            if (fontFaceName[0] != '\0')
+            {
+                htmlBuilder << ConvertToA(CP_UTF8, fontFaceName);
+                htmlBuilder << "',";
+            }
+            // even with different font, add monospace as fallback
+            htmlBuilder << "monospace;";
 
-            // do conversion
-            WideCharToMultiByte(CP_UTF8, 0, wszFontFaceName.data(), static_cast<int>(cchFontFaceName), szBuffer.data(), cbNeeded, nullptr, nullptr);
+            htmlBuilder << "font-size:";
+            htmlBuilder << fontHeightPoints;

```

Comment:
I'm guessing we still need to fix the ambiguous "fontHeightPoints" value? Especially now that the math got cut out, we should make sure it goes into the function after it gets processed?

I've been meaning to get around to this on the other PR, but I've been busy with the accessibility work. Sorry!

Reply:
What's wrong with it? I don't see the ambiguity.

Reply:
There's a comment on #1224 about getting this value. But I think that's outside the scope of this PR, actually. Consider this resolved!